contrib/cindercompatibleapi/api/attachment.go

```diff
@@ -0,0 +1,234 @@
+	"github.com/opensds/opensds/pkg/api/policy"
+	"github.com/opensds/opensds/pkg/model"
+)
+
+// AttachmentPortal ...
+type AttachmentPortal struct {
+	beego.Controller
+}
+
+// DeleteAttachment ...
+func (portal *AttachmentPortal) DeleteAttachment() {
+	if !policy.Authorize(portal.Ctx, "attachment:delete") {
+		return
+	}
+
+	id := portal.Ctx.Input.Param(":attachmentId")
+	attachment := model.VolumeAttachmentSpec{}
+	err := client.DeleteVolumeAttachment(id, &attachment)
+
+	if err != nil {
+		reason := fmt.Sprintf("Delete attachment failed: %v", err)
+		portal.Ctx.Output.SetStatus(model.ErrorBadRequest)
+		portal.Ctx.Output.Body(model.ErrorBadRequestStatus(reason))
+		log.Error(reason)
+		return
+	}
+
+	portal.Ctx.Output.SetStatus(StatusOK)
+	return
+}
+
+// GetAttachment ...
+func (portal *AttachmentPortal) GetAttachment() {
+	if !policy.Authorize(portal.Ctx, "attachment:get") {
+		return
+	}
+
+	id := portal.Ctx.Input.Param(":attachmentId")
+	attachment, err := client.GetVolumeAttachment(id)
+
+	if err != nil {
+		reason := fmt.Sprintf("Show attachment details failed: %v", err)
+		portal.Ctx.Output.SetStatus(model.ErrorBadRequest)
+		portal.Ctx.Output.Body(model.ErrorBadRequestStatus(reason))
+		log.Error(reason)
+		return
+	}
+
+	result := converter.ShowAttachmentResp(attachment)
+	body, err := json.Marshal(result)
+	if err != nil {
+		reason := fmt.Sprintf("Show attachment details, marshal result failed: %v", err)
+		portal.Ctx.Output.SetStatus(model.ErrorInternalServer)
+		portal.Ctx.Output.Body(model.ErrorInternalServerStatus(reason))
+		log.Error(reason)
+		return
+	}
+
+	portal.Ctx.Output.SetStatus(StatusOK)
+	portal.Ctx.Output.Body(body)
+	return
+}
+
+// ListAttachmentsDetails ...
+func (portal *AttachmentPortal) ListAttachmentsDetails() {
+	if !policy.Authorize(portal.Ctx, "attachment:list_details") {
+		return
+	}
+
+	attachments, err := client.ListVolumeAttachments()
+	if err != nil {
+		reason := fmt.Sprintf("List attachments with details failed: %v", err)
+		portal.Ctx.Output.SetStatus(model.ErrorBadRequest)
+		portal.Ctx.Output.Body(model.ErrorBadRequestStatus(reason))
+		log.Error(reason)
+		return
+	}
+
+	result := converter.ListAttachmentsDetailsResp(attachments)
+	body, err := json.Marshal(result)
+	if err != nil {
+		reason := fmt.Sprintf("List attachments with details, marshal result failed: %v", err)
+		portal.Ctx.Output.SetStatus(model.ErrorInternalServer)
+		portal.Ctx.Output.Body(model.ErrorInternalServerStatus(reason))
+		log.Error(reason)
+		return
+	}
+
+	portal.Ctx.Output.SetStatus(StatusOK)
+	portal.Ctx.Output.Body(body)
+	return
+}
+
+// ListAttachments ...
+func (portal *AttachmentPortal) ListAttachments() {
+	if !policy.Authorize(portal.Ctx, "attachment:list") {
+		return
+	}
+
+	attachments, err := client.ListVolumeAttachments()
+	if err != nil {
+		reason := fmt.Sprintf("List attachments failed: %v", err)
+		portal.Ctx.Output.SetStatus(model.ErrorBadRequest)
```

Comment:
BadRequest indicates the improper parameters contained in request, but in this method it is hard to say where the error comes from.